docs/apis/batch/libs/gelly.md

```diff
@@ -2067,7 +2067,92 @@ configuration.
 
   <tbody>
     <tr>
-      <td><strong>TranslateGraphIds</strong></td>
+      <td>degree.annotate.directed.<br/><strong>VertexInDegree</strong></td>
+      <td>
+        <p>Annotate vertices of a directed graph with the in-degree count.</p>
+{% highlight java %}
+DataSet<Vertex<K, LongValue>> inDegree = graph
+  .run(new VertexInDegree()
+    .setIncludeZeroDegreeVertices(true));
+{% endhighlight %}
+      </td>
+    </tr>
+
+    <tr>
+      <td>degree.annotate.directed.<br/><strong>VertexOutDegree</strong></td>
+      <td>
+        <p>Annotate vertices of a directed graph with the out-degree count.</p>
+{% highlight java %}
+DataSet<Vertex<K, LongValue>> outDegree = graph
+  .run(new VertexOutDegree()
+    .setIncludeZeroDegreeVertices(true));
+{% endhighlight %}
+      </td>
+    </tr>
+
+    <tr>
+      <td>degree.annotate.directed.<br/><strong>VertexDegreePair</strong></td>
+      <td>
+        <p>Annotate vertices of a directed graph with both the out-degree and in-degree count.</p>
+{% highlight java %}
+DataSet<Vertex<K, Tuple2<LongValue, LongValue>>> pairDegree = graph
+  .run(new VertexDegreePair()
+    .setIncludeZeroDegreeVertices(true));
+{% endhighlight %}
+      </td>
+    </tr>
+
+    <tr>
+      <td>degree.annotate.undirected.<br/><strong>VertexDegree</strong></td>
+      <td>
+        <p>Annotate vertices of an undirected graph with the degree count.</p>
+{% highlight java %}
+DataSet<Vertex<K, LongValue>> degree = graph
+  .run(new VertexDegree()
+    .setIncludeZeroDegreeVertices(true)
+    .setReduceOnTargetId(true));
+{% endhighlight %}
+      </td>
+    </tr>
+
+    <tr>
+      <td>degree.annotate.undirected.<br/><strong>EdgeSourceDegree</strong></td>
+      <td>
+        <p>Annotate edges of an undirected graph with degree of the source ID.</p>
```

Comment:
When you say "undirected" graph, do you mean the algorithm does not take into account the edge direction or does the input graph have to be undirected? Since gelly graphs are in fact always directed and we simply add opposite-direction edges to represent undirected graphs, we should be clear about this. Also, which is the source vertex is an undirected edge?